### DISCUSSION TITLE
Help needed with setup on MacOS

### DISCUSSION THREAD

**Discussion by David1946** [GerryDazoo/Slinger#70] [Q&A]
Hi I am completely lost when it comes to Python and would appreciate some help. 

So far:-

I have installed Python 3 on my MacMini running MacOS 12.5.1. I have edited the Config.ini file with my password and finderid 

I've also installed a file /etc/sysctl.conf  containing the lines

sysctl -w kern.ipc.maxsockbuf=8192000
sysctl -w net.inet.tcp.sendspace=7250000
sysctl -w net.inet.tcp.recvspace=7250000

I have chosen to run the server on port 8084 as I got an error when I tried using 8080. 

I have set up port forwarding for 8084 and 5303 the port used by my slingbox.

When I run the script it finds the slingbox on my network but then the PING on my external IP Address and port 8084 fails. I am fairly confident that I have set up the port forwarding correctly as I have done it successfully in the past for other devices on my network.

Any help would be much appreciated.

**Comment by GerryDazoo** [comment]
You can comment out your finderid in your config file this will stop the ping. You probably don't need it anyway. But I'm willing to bet even though you have the port forwarded correctly you have not opened the port on your local firewall.
You only need to have the slingbox port forwarding rule in place if you're going to have a server on the Internet connect remote to the slingbox.

**Reply by David1946** [reply]
Many Thanks for the prompt reply. I will try what you have suggested. My slingbox and server are on the same network so I can remove the rule for the slingbox port?

**Reply by David1946** [reply]
It's got a bit further this time but it saying I don't have the correct admin password for the slingbox. Do I copy it literally from the results of searching for my finderids or do I enter what I think my password is - obviously set up a long time ago so perhaps I am not remembering it correctly?

**Reply by GerryDazoo** [reply]
Check my original post for how to get your password. 
Login into you sling account and then go to this web page

"https://newwatchsecure.slingbox.com/watch/slingAccounts/account_boxes_js"

**Reply by David1946** [reply]
Hi I've done that and copied and pasted the adminPassword into my config.ini file but I still get the same result. I've done it including the quotation marks and without but it doesn't make any difference. I feel that I am so close to getting this working remotely and really appreciate your help.

**Reply by GerryDazoo** [reply]
Please post the program output. Your the first person in the world that this hasn't worked for. Maybe the slingbox your connecting too isn't what you think it is. Startup the Sling supplied software and connect. Verify that the network info is the same as what my software is reporting. Do you have more than one slingbox on your account? Also make sure you use a plain jane text editor for editing config.ini.  You may also try doing a quick reset, power off/on the slingbox. Don't do a factory reset.

**Reply by David1946** [reply]
Hi again. Apologies for not getting back to you sooner. Please understand that nobody wants this to work more than I do but as I said in my original post I am not familiar with Python. 

I am currently in Spain and I am working remotely using Google Remote Desktop on a MacMini that is at home and on the same network as the two M1 Slingboxes that I have. Both Slingboxes are working correctly as I can access them both using SlingPlayer on my Phone. I have already installed an H265 encoder and a Broadlink Remote to access the settop box connected to one of the Slingboxes so I am only trying to get Slingbox-Server working for the other Slingbox. When I tried running the script this morning it failed saying that port 8085 is already in use. Here is the output from the program:-
[202209150933.pdf](https://github.com/GerryDazoo/Slinger/files/9573080/202209150933.pdf)
If you need any more info please let me know.

**Reply by David1946** [reply]
Okay I've changed the Server Port back to 8080 and tried again and the Slingbox-server runs successfully as far as I can see however when I try to access the stream with VLC the server says that my admin password is incorrect. According to SlingAccounts my admin password is:-

"adminPassword":"5bp2snbylty4j2m" 

what I have entered in my config.ini file is:-

password=5bp2snbylty4j2m

What am I doing wrong?

**Reply by GerryDazoo** [reply]
The code binds to two ports one you configure in the server section of config.ini. In this case 8084 and it also binds to server port +1 for internal communications. So you need to use something besides 8084 because 8085 is being used by some other program. 

If you want you can run the server in Spain and connect to your slingbox remotely.
On the home router add port forwarding rule to your slingbox.   5303 -> 10.0.101.3:5303
Then in your config.ini in Spain change the IP address to the external IP address of your home router.

PS. I'm wondering if there may still be some threads of a previous run of the server still active. Can you make sure there's no python threads still running? Especially if you keep running into this issue.

**Reply by GerryDazoo** [reply]
David,
It doesn't look wrong. But you said you had two boxes, sorry to state the obvious but are you sure you're using the password for the correct box. Just for grins try the other boxes password.

**Reply by David1946** [reply]
I'm pretty sure I'm using the right one but I will give that a try!

**Reply by David1946** [reply]
You were right that has cured that issue. So we've made good progress, however when I try to access the stream locally using VLC on the MacMini all I get is a black screen. Still can't access it remotely.

**Reply by David1946** [reply]
I'm really pleased to be able to report that I now have it working on the local network. The problem was that I was using the wrong ip address for the box that I wanted. Now I just have to get it working remotely! Possibly a firewall issue.

**Reply by GerryDazoo** [reply]
The black screen is usually due to selecting a VideoSource that has no input. The config.ini has some rudimentary documentation on what source to set for each slingbox type.
A firewall issue is very likely. Many people forget that, thinking the router port forwarding rule is sufficient.

**Reply by David1946** [reply]
Thanks for all your help I really appreciate it and many many thanks for developing this solution for the SlingBox.

**Reply by David1946** [reply]
I'm very pleased to report that I have got it working by setting up the server here on my MacBook. Just have to sort out the remote although I may be tempted to opt for a Broadlink RM Mini 3.
